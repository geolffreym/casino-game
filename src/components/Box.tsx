import React, { FC, PropsWithChildren } from 'react'
import { WithTheme, withTheme } from '@material-ui/core'
import styled from 'styled-components'

/* eslint-disable  @typescript-eslint/strict-boolean-expressions */
const Box: FC<PropsWithChildren<{ title?: string }>> = ({ title, children }): JSX.Element => {
  return (
    <BoxWrapper>
      {
        title &&
          <BoxHeader>{title}</BoxHeader>
      }
      <BoxContent> {children} </BoxContent>
    </BoxWrapper>
  )
}

export default Box

export const BoxHorizontalWrapper = styled.div`
  width: 100%;
  display: flex;
  flex-wrap: wrap;

  & > div {
    margin-right: 1rem;
    max-width: calc((100% / 3) - (2rem / 3));

    &:nth-of-type(3n) {
      margin-right: 0;
    }
  }

  @media (min-width: 1900px) {
    & > div {
      margin-right: 1rem;
      max-width: calc((100% / 4) - (3rem / 4));

      &:nth-of-type(4n) {
        margin-right: 0;
      }
    }
  }

  @media (max-width: 992px) {
    & > div {
      margin-right: 1rem;
      max-width: calc((100% / 2) - 1rem);

      &:nth-of-type(2n) {
        margin-right: 0;
      }
    }
  }

  @media (max-width: 640px) {
    & > div {
      margin-right: 0;
      max-width: 100%;
    }
  }
`

export const BoxWrapper = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  background-color: #fff;
  box-shadow: 2px 2px 8px rgba(0, 0, 0, 0.15);
  border-radius: 1rem;
  margin-bottom: 1rem;
  overflow: hidden;
`

export const BoxHeader = withTheme(
  styled.div<WithTheme>`
    width: 100%;
    display: flex;
    padding: 1rem;
    color: ${({ theme }) => theme.palette.primary.main};
    font-size: 1.1rem;
    font-weight: bold;
  `
)

const BoxContent = styled.div`
  width: 100%;
`
