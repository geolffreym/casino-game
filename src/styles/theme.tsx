// Global styles can be moved to a separate file for ease of maintenance.
export const globalOverrides = {
  overrides: {
    MuiCssBaseline: {
      '@global': {
        html: {
          margin: 0,
          padding: 0,
          height: '100%'
        },
        body: {
          margin: 0,
          padding: 0,
          height: '100%',
          fontSize: '1rem',
          backgroundColor: '#fff'
        }
      }
    }
  }
}

export const typography = {
  typography: {
    fontFamily: [
      'Tamil Sangam MN',
      'Oswald',
      'Arial',
      'sans-serif'
    ].join(',')
  }
}

export const defaultTheme = {
  palette: {
    primary: {
      light: '#bf1f34',
      main: '#990015',
      dark: '#750212',
      contrastText: '#E4F1F0'
    },
    secondary: {
      main: '#2795B7',
      contrastText: '#fff'
    },
    warning: {
      light: '#FEF5B2',
      main: '#C79C36',
      dark: '#814E22'
    },
    success: {
      light: '#DEF7EC',
      main: '#3C763D',
      dark: '#2D6C59'
    },
    default: {
      main: '#fff',
      contrastText: '#9999'
    },
    grey: {
      50: '#F2F6FF',
      150: '#F9FAFC',
      200: '#E6EBF1',
      250: '#98A5BA'
    }
  }
}
