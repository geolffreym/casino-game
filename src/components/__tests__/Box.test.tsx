import React from 'react'
import { mount, shallow } from 'enzyme'
import Box, { BoxHorizontalWrapper, BoxHeader, BoxWrapper } from '@components/Box'
import { ThemeProvider, createTheme } from '@material-ui/core/styles'
import { defaultTheme } from '@styles/theme'
import { Theme } from '@material-ui/core'
import 'jest-styled-components'

/* eslint-disable no-undef */
/* eslint-disable @typescript-eslint/consistent-type-assertions */
describe('<Box />', () => {
  let theme: Theme

  beforeAll(() => {
    theme = createTheme(defaultTheme)
  })

  it('should render', () => {
    const component = shallow(<Box title='Test Module' />)
    expect(component).toMatchSnapshot()
  })

  it('should pass if box header exist in module', () => {
    const component = shallow(<Box title='Test Module' />)
    const header = component.find(BoxHeader)

    expect(header.exists()).toBeTruthy()
  })

  it('should fail if box header exist in module', () => {
    const component = shallow(<Box />)
    const header = component.find(BoxHeader)

    expect(header.exists()).toBeFalsy()
  })

  it('should have box props text', () => {
    const titleText = 'Test Module'
    const component = shallow(<Box title={titleText} />)
    const headerText = component.find(BoxHeader)

    expect(headerText.text()).toEqual(titleText)
  })

  it('should have box horizontal wrapper valid styles', () => {
    const component = shallow(<BoxHorizontalWrapper />)

    expect(component).toHaveStyleRule('width', '100%')
    expect(component).toHaveStyleRule('display', 'flex')
    expect(component).toHaveStyleRule('flex-wrap', 'wrap')
  })

  it('should have box wrapper valid styles', () => {
    const component = shallow(<BoxWrapper />)

    expect(component).toHaveStyleRule('width', '100%')
    expect(component).toHaveStyleRule('display', 'flex')
    expect(component).toHaveStyleRule('flex-direction', 'column')
    expect(component).toHaveStyleRule('box-shadow', '2px 2px 8px rgba(0,0,0,0.15)')
  })

  it('should have box header valid theme styles', () => {
    const boxHeader = mount(
      <ThemeProvider theme={theme}>
        <BoxHeader />
      </ThemeProvider>
    )

    expect(boxHeader).toHaveStyleRule('color', theme.palette.primary.main)
    expect(boxHeader).toHaveStyleRule('color', '#990015')
  })
})
