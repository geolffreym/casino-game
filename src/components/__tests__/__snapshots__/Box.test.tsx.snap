// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> should render 1`] = `ShallowWrapper {}`;
